core/controllers/topic_editor.py

```diff
@@ -224,6 +227,37 @@ def delete(self, topic_id):
         topic_services.delete_topic(self.user_id, topic_id)
 
 
+class TopicRightsHandler(base.BaseHandler):
+    """A handler for returning topic rights."""
+
+    @acl_decorators.can_manage_rights_for_topic
+    def get(self, topic_id):
+        """Returns the TopicRights object of a topic.
+        """
+        topic_domain.Topic.require_valid_topic_id(topic_id)
+
+        try:
+            topic_rights = topic_services.get_topic_rights(topic_id)
+            user_actions_info = user_services.UserActionsInfo(self.user_id)
+            can_edit_topic = False
+            if topic_services.check_can_edit_topic(
+                    user_actions_info, topic_rights):
+                can_edit_topic = True
+        except Exception as e:
+            raise self.PageNotFoundException(e)
+
+        topic_rights_to_return = {
+            'topic_id': topic_id,
+            'can_edit_topic': can_edit_topic,
+            'is_published': topic_rights.topic_is_published
+        }
+        self.values.update({
+            'topic_rights': topic_rights_to_return
```

Comment:
I'm not sure about the need for the extra indirection. Perhaps just return the dict with topic_id, can_edit_topic, etc. instead?

Also, not sure why we need to send the topic_id back. Presumably the frontend already knows it.

Reply:
Ok, yeah, removed topic_id, done.